### ISSUE TITLE
Improve project root detection

### ISSUE THREAD

**Issue by kentcdodds** [gtramontina/ghooks#3]
Looks like in the `install.js` script it's looking for the `node_modules` directory and calling that root. Couldn't it look for the `.git` directory instead? Or am I understanding this wrong?

**Comment by DrewML** [comment]
@kentcdodds I believe the errors you got trying to use a 3rd party module in an install/postinstall script were due to https://github.com/npm/npm/issues/6926 (fixed about 2 months ago). It might be possible now to get this done, without having to copy/paste a ton of code from a 3rd party module.

I'm very interested in this issue. I'd like to use `ghooks` for a project at work, but the `node_modules` directory lives several subdirectories under the root of the git repo.

I may work on a PR for this once I get some sleep :)

**Comment by DrewML** [comment]
@gtramontina What is the minimum node version that you wish to support for this module?

It looks like the 2 modules you suggested above (exec-sync and execSync) are both no longer maintained. However, io.js and node.js >= 0.12 both [now have a method in the core `child_process` lib to do this](https://nodejs.org/api/child_process.html#child_process_child_process_execsync_command_options)

**Comment by kentcdodds** [comment]
Not my lib, but I'd be fine with that change.

**Comment by DrewML** [comment]
We could also feature detect for `child_process.execSync`, and fall back to one of those modules if it's not present. However, I don't know that it'd be worth it.

It looks like the newer of those 2 modules ([execSync](https://github.com/mgutz/execSync)), has a dependency on Python and Visual Studio for Windows users. And, since it's no longer maintained, it may just make life more difficult for users that try to use this lib on node < 0.12

**Comment by DrewML** [comment]
I have most of this done in a branch, but I did run into one issue that I'm not sure of a solution for.

Summary of the problem:
1. `hook.template.raw` is copied to each file in `.git/hooks`
2. Each file in `.git/hooks` requires `ghooks`
3. Since `.git` is in the top-level directory, but the `node_modules` folder is 1 or several sub-directories deep, node's `require` resolution does not find this module.

tl;dr: We need to determine a way for each hook file to discover the path to the `node_modules` directory.

**Comment by DrewML** [comment]
https://github.com/DrewML/ghooks/tree/issue3

I now have it working so you can have your `package.json` at any folder depth. All tests are also passing.

I have one more bug I need to address. Because of an assumption my code makes about fs structure, this only works if it lives in a `node_modules` directory (meaning it works everywhere *but* this repository, ironically).

**Comment by gtramontina** [comment]
Hey there @DrewML, sorry for the silence... Thank's a lot for taking a stab at this issue :beers: I really don't mind releasing a `ghooks 2.0` version which requires node `0.12`, or `iojs` for that matter. Using semver for what it is meant for... :smile:

**Comment by DrewML** [comment]
Excellent -- thanks @gtramontina. PR submitted.

**Comment by kentcdodds** [comment]
This is related to #28. Would love to knock them both out.

**Comment by kentcdodds** [comment]
I recommend we follow what @noamokman recommends [here](https://github.com/gtramontina/ghooks/pull/1#issuecomment-155849226)

**Comment by gtramontina** [comment]
Closing this one, as #37 addresses it.
